src/tools/idlc/src/idl.c

```diff
@@ -171,7 +181,7 @@ int32_t idl_parse(idl_processor_t *proc)
   do {
     if ((code = idl_scan(proc, &tok)) < 0)
       break;
-    if (proc->state & IDL_SCAN_DIRECTIVE)
+    if ((int)proc->state & (int)IDL_SCAN_DIRECTIVE)
```

Comment:
Bitwise operations are on `unsigned`s so I think I would cast to `unsigned` rather than `int` — it doesn't change the behaviour. I would consider defining an inline function or a macro next to the enum definition that tests whether it is in "directive scanning" mode, but I can live with this, too.